internal/configs/version2/nginx-plus.virtualserver.tmpl

```diff
@@ -68,8 +68,17 @@ match {{ $m.Name }} {
 {{ end }}
 
 server {
+
+    {{ if $.InternalRouteServer }}
+    listen 443 ssl;
+    {{if not $s.DisableIPV6}}listen [::]:443 ssl;{{end}}
+    ssl_certificate /etc/nginx/secrets/spiffe_cert.pem;
+    ssl_certificate_key /etc/nginx/secrets/spiffe_key.pem;
```

Comment:
This is a separate server block in `Ingress`...don't we need to do the same here? And anyway I don't think it should "remove" the listen on port 80?

Reply:
In ingress it also switches off different features. https://github.com/nginxinc/kubernetes-ingress/blob/da4566c4e3a09a6387aa339066b7ba2a70bd1d9a/internal/configs/version1/nginx-plus.ingress.tmpl#L23

Does port 80 provide functionality to the internal route that I'm switching off? If so I can adjust the logic.

Reply:
I guess I got confused because in ingress the same logic is for `SpiffeCerts` instead of `InternalRouteServer`...right?

Reply:
Yeah, I agree that was a bit confusing. But yes it's the same logic that `SpiffeCerts` provides.

Reply:
shouldn't we be using `SpiffeCerts` then?

Reply:
It's been a while so I had to do some digging. Doesn't this have to do with the way vs and ingress diverge? We follow the similar logic for ingress here: https://github.com/nginxinc/kubernetes-ingress/blob/da4566c4e3a09a6387aa339066b7ba2a70bd1d9a/internal/configs/version1/nginx.tmpl#L213

Reply:
Spoke with Luca offline and provided clarification.